docs/get-started/getting-started-kubernetes.md

```diff
@@ -0,0 +1,256 @@
+   ```shell 
+   export IMAGE_NAME=xxx/yyy:tag
+   docker build . -f Dockerfile.cuda -t $IMAGE_NAME
+   docker push $IMAGE_NAME
+   ```
+
+## Run Spark Applications in Kubernetes Cluster
+
+### spark-submit example pi job in `cluster` mode.
+
+This is a good way to test if RAPIDS plugin can be found.
+Because `ClassNotFoundException` is a common error if the Driver can not 
+find the jar for `com.nvidia.spark.SQLPlugin`:
+```
+Exception in thread "main" java.lang.ClassNotFoundException: com.nvidia.spark.SQLPlugin
+```
+
+Here is an example job: 
+
+```shell
+export SPARK_HOME=~/spark
+export IMAGE_NAME=xxx/yyy:tag
+export K8SMASTER=k8s://https://<k8s-apiserver-host>:<k8s-apiserver-port>
+export SPARK_NAMESPACE=default
+export SPARK_DRIVER_NAME=exampledriver
+
+$SPARK_HOME/bin/spark-submit \
+     --master $K8SMASTER \
+     --deploy-mode cluster  \
+     --name examplejob \
+     --class org.apache.spark.examples.SparkPi \
+     --conf spark.rapids.sql.concurrentGpuTasks=1 \
+     --conf spark.executor.instances=1 \
+     --conf spark.executor.resource.gpu.amount=1 \
+     --conf spark.executor.memory=4G \
+     --conf spark.executor.cores=1 \
+     --conf spark.task.cpus=1 \
+     --conf spark.task.resource.gpu.amount=1 \
+     --conf spark.rapids.memory.pinnedPool.size=2G \
+     --conf spark.locality.wait=0s \
+     --conf spark.sql.files.maxPartitionBytes=512m \
+     --conf spark.sql.shuffle.partitions=10 \
+     --conf spark.plugins=com.nvidia.spark.SQLPlugin \
+     --conf spark.kubernetes.namespace=$SPARK_NAMESPACE  \
+     --conf spark.kubernetes.driver.pod.name=$SPARK_DRIVER_NAME  \
+     --conf spark.executor.resource.gpu.discoveryScript=/opt/sparkRapidsPlugin/getGpusResources.sh \
+     --conf spark.executor.resource.gpu.vendor=nvidia.com \
+     --conf spark.kubernetes.container.image=$IMAGE_NAME \
+     --conf spark.executor.extraClassPath=/opt/sparkRapidsPlugin/rapids-4-spark_<version>.jar:/opt/sparkRapidsPlugin/cudf-<version>.jar   \
+     --conf spark.driver.extraClassPath=/opt/sparkRapidsPlugin/rapids-4-spark_<version>.jar:/opt/sparkRapidsPlugin/cudf-<version>.jar   \
+     --driver-memory 2G \
+     local:///opt/spark/examples/jars/spark-examples_2.12-3.0.2.jar
+```
+
+   Note: `local://` means the jar file location is inside the Docker image.
+   Since this is `cluster` mode, the Driver is running inside a POD in Kubernetes.
+   When the job is running, we could see both Driver POD and Executor POD(s):
+```shell
+$ kubectl get pods
+NAME                               READY   STATUS    RESTARTS   AGE
+spark-pi-d11075782f399fd7-exec-1   1/1     Running   0          9s
+exampledriver                      1/1     Running   0          15s
+```
+
+   To view the Driver log, use below command:
+```shell
+kubectl logs $SPARK_DRIVER_NAME
+```
+
+   To view the Spark Driver UI when the job is running:
+```shell
+kubectl port-forward $SPARK_DRIVER_NAME 4040:4040
```

Comment:
would this be cluster specific?

Reply:
This is not. "4040" is the default port for driver and is also used by Spark doc: https://spark.apache.org/docs/latest/configuration.html

Reply:
what is multiple drivers on the same host?  Spark increments the port number for you.   Do some clusters just have aces open so I can just access it. etc... this is what I mean by cluster specific. I'm fine with this as long as we document.

Reply:
Oh i see, i think configurations like spark.driver.port should not need to be documented by us and users can refer to spark guide for all configs. this may just be a quick start guide. How do you think?

Reply:
yeah quick start is fine perhaps at the beginning of the doc we can add a sentence ssaying this is quick start thing can be slightly different for different cluster setups, see your spark configs.

Reply:
"This is a quick start guide which uses default settings which may be different from your cluster." added.